src/S.res

```diff
@@ -488,8 +487,8 @@ type taggedLiteral =
   | NaN
 type operation =
   | NoOperation
-  | SyncOperation((. unknown) => unknown)
-  | AsyncOperation((. unknown) => (. unit) => promise<unknown>)
+  | SyncOperation(unknown => unknown)
+  | AsyncOperation(unknown => unit => promise<unknown>)
```

Comment:
Looks like `rescript format` was run with `uncurried: true`. This might work incorrectly in a curried mode.

Reply:
Interesting, I did run rescript format with the correct setting. But is seems like with this version of rescript it removes the `uncurried` notation. I'll double check.

It does seem like there will be some backwards compatibility issues with this version of rescript...

Reply:
If so, it looks like it doesn't take the `uncurried: false` in `bsconfig` into consideration and should be reported to the compiler repo.

Reply:
Agreed!

Reply:
https://github.com/rescript-lang/rescript-compiler/issues/6307